caddyhttp/proxy/reverseproxy.go

```diff
@@ -261,7 +267,10 @@ func (rp *ReverseProxy) copyResponse(dst io.Writer, src io.Reader) {
 			dst = mlw
 		}
 	}
-	io.CopyBuffer(dst, src, buf.([]byte))
+
+	// `CopyBuffer` only uses `buf` up to it's length and
+	// panics if it's 0 => Extend it's length up to it's capacity.
```

Comment:
This comment is hard to understand. Can it be phrased a little more clearly (and fix the gammar, haha)? Maybe something like:

> CopyBuffer only uses buf up to its length and panics if it's 0, so we extend its length up to its capacity.